### ISSUE TITLE
Atom Editor Doesn't Come to Foreground

### ISSUE THREAD

**Issue by cagataycebi** [atom/atom#6201]
Hi,

My OS is Ubuntu 14.10. I use Unity as my window manager. Atom version is 0.188.0. 

If I open a file from Nautilus file manager by pressing "Enter" or by double-clicking, Atom often doesn't come to foreground. (Its icon wobbles in Unity dashboard.) To see it, I have to switch to Atom manually. Only after that, I can work on the file. 

I made some tests. It happens when Atom is already open and if I use Alt+Tab key combination. If I never use Alt+Tab to switch between windows, then there's no problem. Otherwise, first I've to open file, then manually switch to Atom window. Unfortunately It's inefficinet way to work.

What I want is, Atom should always come to foreground, if I open a file. Is there a workaround or a patch for this problem?

Kind regards,
Cagatay

**Comment by cagataycebi** [comment]
I found a similar issue:  "Atom editor does not go to foreground when opened from console #4324". I believe both issues can be solved with a same fix.

**Comment by 50Wliu** [comment]
I've noticed this as well.

**Comment by mnquintana** [comment]
:+1: I've seen this too, on Ubuntu 14.04 and Atom 0.188.0

**Comment by mishak87** [comment]
same

**Comment by Enkouyami** [comment]
I don't have this issue with the cinnamon DE, so it could just be a Unity bug and not an atom bug.

**Comment by cagataycebi** [comment]
@Enkouyami, maybe you're right. But all other programs works as expected. For example if I open a file in any other text editor (such as Sublime, Geany, Gedit etc...), then they always come to foreground. So it should be related to Atom. 

BTW, I narrowed the issue. If I open Atom and click any key within the editor, this problem occurs. Let me give an example: Open Atom and write "Hello World". If you try to open another file, Atom will not come to foreground.

**Comment by rezzzet** [comment]
+1 on Ubuntu 14.10 x64

**Comment by jhasse** [comment]
I also see this bug with other programs, not Atom only.

Also see this (pretty old) bug: https://bugzilla.gnome.org/show_bug.cgi?id=429124

**Comment by samuelgzhang** [comment]
similar problem found on my windows 7 with atom 0.208.0 version.

**Comment by pandrei** [comment]
I have tested this behavior on an Ubuntu 14.10 x64 and indeed it does happen that way,however I have tried other interfaces(on the same OS), like kubuntu-desktop and the bug does not reproduce. 
This seems to be Unity specific, caused by Unity Dash. If you can provide more details I can investigate that case.

**Comment by Pixy** [comment]
Any news on this bug ? 
I also have it on Win8 and Win7, and makes the editor unusable for me =(

**Comment by jhasse** [comment]
I think this is a duplicate of #4324

**Comment by nige123** [comment]
Still a problem on Ubuntu.

**Comment by msiniscalchi** [comment]
Still a problem on Windows, with Atom 1.3.2.

**Comment by Nek80** [comment]
Still a problem here too. Windows 10 x86_64 Atom 1.3.3

**Comment by Zireael07** [comment]
@Nek80 There's 1.4.0 out already!

**Comment by Nek80** [comment]
@Zireael07 Yay!! I just installed it. But... still the same problem with 1.4.0 too. (windows 10 x64)

**Comment by moio** [comment]
Still present in 1.4.0 with Ubuntu 14.04.

**Comment by DKhalil** [comment]
Still an issue here on Win10 with 1.7.2

**Comment by ArniPL** [comment]
Issue still present. Is there any news on this? Windows 10, Atom 1.8.0.

**Comment by ArniPL** [comment]
I've found that it only happens (Windows), when Atom window is *not* minimized. If you open a file and Atom is minimized, then everything works as expected.

**Comment by robeerob** [comment]
Same issue here. I found this issue (closed) to be describing the issue even better (linking in case it helps to solve it):
https://github.com/atom/atom/issues/5938

I am on Ubuntu 16.04.1 with Unity, and Atom 1.5.4.  I can reproduce the issue when I use the "Restore window" keyboard shortcut to change the size of the window from maximized to smaller. Not sure that is how it happened originally.

**Comment by Keith94** [comment]
Any progress on fixing this important bug?

**Comment by zBernie** [comment]
Running Atom 1.14.4 on Windows 10,  files opened in Explorer do not come to the foreground.  But on a Windows 7 and Mac computer, Atom does get focus when opening a file from Explorer or Finder.

**Comment by zBernie** [comment]
Okay, I just tried again, and it's working!  If I right click on a file, then select "Open in Atom" from the context menu, Atom gets focus and the is brought to the foreground!  I've tried this numerous times over the past week and it did not work.  I'm using version 1.14.4 x64.

**Comment by ArniPL** [comment]
Tested again on 1.14.4 - bug still present. Happens only when **not** minimized (when another window is in the foreground and Atom is in the background behind that window). See [this comment](https://github.com/atom/atom/issues/6201#issuecomment-224524737) above.

@zBernie Could you please test again if you're experiencing the same behavior?

**Comment by zBernie** [comment]
I cannot get this to fail now if I try!  I tried exactly what you suggested, I tried with Atom minimized, and not minimized, and now whenever I right click on a file and select "Open in Atom" from the context menu, Atom comes to the front and gets focus!  This is with version 1.14.4 x64.  I'm almost positive I had the focus issue with this version.  But now it works.  Maybe Atom is not the problem, and a Windows update resolved it?

**Comment by math0ne** [comment]
I'm still experiencing this in 1.18.0 x64 in windows 10.

**Comment by wedi** [comment]
I'm still experiencing this in 1.19.0 x64 in windows 10.

**Comment by arssonist** [comment]
Experiencing this with win7 opening from anywhere. Atom will never open in foreground and I MUST click the taskbar to enter the window. There is no way around it.

**Comment by earonesty** [comment]
i often run "atom filename" from the command line, and it does not bring to front. (1.20.0 x 64, windows 10), although it does begin "blinking" when i do this... so i know some sort of event was received by the program

**Comment by dnknitro** [comment]
Is somebody working on this bug (on Windows 10)? It is very user unfriendly :(

**Comment by RaynorD** [comment]
Still experiencing this in 1.27.2 x64 in windows 10.

**Comment by clagms** [comment]
I'm running version 1.30.0 x64 on Windows 10 Home, and this is still an issue.

**Comment by ps374** [comment]
I am also experiencing this issue: [https://github.com/atom/atom/issues/4324#issuecomment-445737826](https://github.com/atom/atom/issues/4324#issuecomment-445737826)

**Comment by ps374** [comment]
It worked this time :) 

I am not sure how but I had just deleted the **.atom-backup** directory from **C:\Users\<USER>** profile folder and then run **Atom** as Administrator that worked like a charm.

Lovely :)

**Comment by debanshumukherjee** [comment]
I got this resolved by adding my atom.exe file path to the 'path' environment variable in the advanced system setting (Control Panel\System and Security\System)

**Comment by kendonB** [comment]
> I got this resolved by adding my atom.exe file path to the 'path' environment variable in the advanced system setting (Control Panel\System and Security\System)
This worked for me on Windows 10 x64.

**Comment by kendonB** [comment]
@debanshumukherjee did you also find a workaround to my problem ^?

**Comment by shayanderson** [comment]
Problem still exists on Win 10 Pro v1909 Atom v1.44.0 x64. I experience this issue on multiple laptops. Atom is unusable with this issue.

**Comment by Hativ** [comment]
5 years and still an issue :-(

I have the same problem on Ubuntu 20.04 LTS with Atom 1.49.0.

If I have opened a file and minmized Atom and try to open a new file Atom doesn't come to foreground. It just comes to foreground when behind another window.

**Comment by pasztorpisti** [comment]
I solved my issue on macOS Catalina by downgrading Atom from 1.53.0 to 1.51.0.

I always install atom manually for my user accounts to `~/Applications/Atom.app` (and create a `~/bin/atom` link pointing to `~/Applications/Atom.app/Contents/Resources/app/atom.sh`) and turn off auto-updates (a checkbox on the "About" page/tab) to avoid running into these recurring atom/electron bugs. Unfortunately on a new user account I forgot to turn off Atom's auto-updater so it downloaded a few annoying bugs.

The latest version (1.53.0) couldn't reopen the atom window on login so I downgraded to 1.52.0. In that version I immediately noticed that it couldn't bring the window to the foreground in case of opening a file from the terminal with my `atom` link/command. I ended up with version 1.51.0 that can reopen the window on login and can bring the window to the foreground from the commandline.

Note: I didn't check whether 1.53.0 is capable of bringing the window to the foreground in case of opening a file from the commandline and didn't check whether 1.52.0 is able to reopen the window after logging in.

These annoying "window management" related bugs in 1.53.0 and 1.52.0 aren't so surprising because the release note of 1.52.0 mentions "Upgrade to electron 6".

**Comment by Luklek** [comment]
Guess it's time to go back to Notepad++
5 years and nobody took care of such an important bug speaks volume of the devs here...

**Comment by caseydwayne** [comment]
Bah-bum-bump. Win 10, same problem. Getting tired of closing explorer windows instead of the file I wanted to close in Atom...

**Comment by kaicode** [comment]
Found MacOS workaround... Still seeing this issue when installed by Homebrew on latest MacOS (12.0.1) with latest Atom (1.58.0).

**Workaround**: Remove the brew version of atom and install using directly from https://atom.io/ (unzip, drag to Applications).
Atom now comes to front when opening files on the command line. ☺️

**Comment by eli-hu** [comment]
Still a problem, Windows 10, Atom 1.60.0 x64.
My workflow:
1. In Windows Explorer, double-click a pdf to open in Acrobat. The pdf opens and Acrobat gains focus.
2. In Windows Explorer, double-click a txt, md, yml, or any other file that's configured with Atom as the default handler. The file opens but Atom doesn't gain focus.
